packages/design-system/src/utils/event-bus.ts

```diff
@@ -2,13 +2,51 @@
 export type CallbackFn = Function;
 export type UnregisterFn = () => void;
 
-export interface EventBus {
-	on: (eventName: string, fn: CallbackFn) => UnregisterFn;
-	off: (eventName: string, fn: CallbackFn) => void;
-	emit: <T = Event>(eventName: string, event?: T) => void;
+export type Listener<Payload> = (payload: Payload) => void;
+
+export type Payloads<ListenerMap> = {
+	[E in keyof ListenerMap]: unknown;
+};
+
+// TODO: Fix all usages of `createEventBus` and convert `any` to `unknown`
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface EventBus<ListenerMap extends Payloads<ListenerMap> = Record<string, any>> {
```

Comment:
instead of keeping this as an interface, could we make this a class, and move the method implementations in here. Then  `createEventBus` could just create an instance of the class.

Reply:
Yes let's do that

Reply:
reverted back to `interface`, but with less code duplication.